feat: add onboarding mnemonic and congratulation views for mobile stardust

paul-boegelsack opened the pull request:
## Summary

This PR adds the mnemonic views as well as the congratulation screen to the mobile stardust onboarding.

## Changelog

```
- Adds profiel backup router
- Adds recovery phrase component
- Adds view mnemonic screen to onboarding
- Adds verify mnemonic screen to onbarding
```

## Relevant Issues

Closes #4625
Closes #4626

## Testing

Until the mobile bindings are ready, the `dev-mode/mobile-stardust` branch has to be utilized for testing:

1 `git checkout dev-mode/mobile-stardust` 
2 `git merge feat-mobile/new-onboarding-mnemonic-views`
3 `yarn (root directory)`
4 `yarn build (packages/desktop)`
5 `yarn start (packages/desktop)`

press `ctrl + c` to change the color theme
press `ctrl + k` to toggle the keyboard

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [ ] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

begonaalvarezd commented on packages/mobile/components/RecoveryPhrase.svelte:
```diff
@@ -0,0 +1,47 @@
+            filter: blur(4px);
+        }
+        span {
+            &.errored {
```
In figma it shows how we should always highlight the current word you are verifying, and if wrong then it red be red
![image](https://user-images.githubusercontent.com/3624944/193814012-e36a9dd5-7508-43f6-9250-c4986327ad44.png)
![image](https://user-images.githubusercontent.com/3624944/193814084-97da1e5d-08a2-42f8-afac-b1f5a2eb192b.png)

begonaalvarezd replied:
Though we can polish this later on in a polish PR

begonaalvarezd commented on packages/mobile/routes/onboarding/views/profile-backup/views/VerifyMnemonicView.svelte:
```diff
@@ -0,0 +1,134 @@
+        $profileBackupRouter.previous()
+    }
+
+    function onKeyPress(event: KeyboardEvent): void {
```
I think we could remove this for mobile

begonaalvarezd commented on packages/mobile/routes/onboarding/views/profile-backup/views/VerifyMnemonicView.svelte:
```diff
@@ -0,0 +1,134 @@
+    })
+</script>
+
+<svelte:window on:keypress={onKeyPress} />
```
same with this, we could remove it

begonaalvarezd reviewed (changes requested):
Looking amazing 🔥 Great work 🚀 
Just added some comments 🙏🏼

begonaalvarezd commented on packages/mobile/routes/onboarding/views/CongratulationsView.svelte:
```diff
@@ -0,0 +1,35 @@
+                >{localize('views.onboarding.congratulations.body')}</Text
+            >
+        </div>
+        {#if $isOnboardingLedgerProfile}
```
I guess we could also remove this?

paul-boegelsack commented:
> Looking amazing 🔥 Great work 🚀 Just added some comments 🙏🏼

Thank you 🙏  hopefully I addressed everything! I changed the UI feedback of the verification regarding figma a bit more.

begonaalvarezd reviewed (approved):
🔥 neat
